controllers/pipelines/runconfiguration_controller.go

```diff
@@ -78,11 +77,11 @@ func (r *RunConfigurationReconciler) Reconcile(ctx context.Context, req ctrl.Req
 	provider := runConfiguration.Spec.Run.Provider
 	if !runConfiguration.Status.Provider.Empty() && provider != runConfiguration.Status.Provider {
 		//TODO: refactor to use Commands and introduce a StateHandler
-		runConfiguration.Status.SynchronizationState = apis.Failed
+		runConfiguration.Status.SetSynchronizationState(apis.Failed, StateHandlerConstants.ProviderChangedError)
 
 		message := fmt.Sprintf(
 			`%s: %s`,
-			string(runConfiguration.Status.SynchronizationState),
+			string(apis.Failed),
 			StateHandlerConstants.ProviderChangedError,
 		)
 		r.EC.Recorder.Event(runConfiguration, EventTypes.Warning, EventReasons.SyncFailed, message)
```

Comment:
Line 84: Should we read the synchronization state (`apis.Failed`) from the conditions instead of hardcoding it again? Or is there a reason we didn't do that?

Reply:
We set it above to be apis.Failed, extracting this from the list of conditions is wasted effort when we have just explicitly set this to be failed.

Reply:
Fair enough, had the same thought - just thought I'd point it out.

I was actually mentioning it because there's a 3rd time below that it's done, but I realised that one's not fixable.